src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2301,27 +2300,16 @@ private static ClassFile readClassFile(byte[] bytes) {
                 String name;
                 int accessFlags;
                 try {
-                    ClassReader reader = new ClassReader(bytes);
-                    // ClassReader does not check if `this_class` is CONSTANT_Class_info
-                    // workaround to read `this_class` using readConst and validate the value
-                    int thisClass = reader.readUnsignedShort(reader.header + 2);
-                    Object constant = reader.readConst(thisClass, new char[reader.getMaxStringLength()]);
-                    if (!(constant instanceof Type type)) {
-                        throw new ClassFormatError("this_class item: #" + thisClass + " not a CONSTANT_Class_info");
-                    }
-                    if (!type.getDescriptor().startsWith("L")) {
-                        throw new ClassFormatError("this_class item: #" + thisClass + " not a CONSTANT_Class_info");
-                    }
-                    name = type.getInternalName();
-                    accessFlags = reader.readUnsignedShort(reader.header);
+                    ClassModel cm = Classfile.parse(bytes);
+                    name = cm.thisClass().asInternalName();
+                    accessFlags = cm.flags().flagsMask();
                 } catch (RuntimeException e) {
```

Comment:
We can now anticipate an IllegalArgumentException after our recent exception consolidation in Classfile API.

Reply:
Fixed, thanks.